src/core/core/task_engine.h

```diff
@@ -76,6 +76,7 @@ class task_worker_pool
     void get_runtime_info(const std::string& indent, const std::vector<std::string>& args, /*out*/ std::stringstream& ss);
     std::vector<task_queue*>& queues() { return _queues; }
     std::vector<task_worker*>& workers() { return _workers; }
+    std::vector<admission_controller*>& controllers() { return _controllers; }
```

Comment:
ok, let me do the cleanup work later. the same will be applied to rpc_replicated.